### PR TITLE
Create first test 'Computing precision set to 3' for UI tests new campaign  'Regression'

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Create first test 'Computing precision set to 3' for UI tests new campaign  'Regression' (For now this campaign will not run during the nightly)
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `URL_FO=shopUrl npm run regression-tests`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21327)

### CODE REVIEW

File: tests/UI/campaigns/data/faker/cartRule.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ module.exports = class CartRule {
     this.quantityPerUser = cartRuleToCreate.quantityPerUser || 1;
 
     // Actions
-    this.freeShipping = cartRuleToCreate.freeShipping === undefined ? 'on' : cartRuleToCreate.freeShipping;
+    this.freeShipping = cartRuleToCreate.freeShipping === undefined ? false : cartRuleToCreate.freeShipping;
```
Comment (SimonGrn) [inline tests/UI/campaigns/data/faker/cartRule.js:31]: How did this work until now ?

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule.js [hunk 2]
```diff
@@ -148,9 +148,9 @@ describe('CRUD cart rule', async () => {
     });
 
     it('should set the promo code', async function () {
-      await testContext.addContextItem(this, 'testIdentifier', 'setPromoCode_1', baseContext);
+      await testContext.addContextItem(this, 'testIdentifier', 'addPromoCode_1', baseContext);
```
Comment (SimonGrn) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule.js:151]: I understand but modifying this file seems to be out of scope, since it's in the `functional` campaign ?

File: tests/UI/campaigns/regression/currencies/computingPrecision_FO.js [hunk 3]
```diff
@@ -0,0 +1,535 @@
...
+/*
+Change euro decimal to 3
+Create 2 cart rules
+  1. 15% discount
+  2. Product 'Mug today is a good day' as a gift product
+Place an order in FO with the cart rules created
+  1. Check discount value after first cart rule added
+  2. Check discount value after second cart rule added
+  3. CheckTTC after second cart rule added
+  4.Finish the order
```
Comment (SimonGrn) [inline tests/UI/campaigns/regression/currencies/computingPrecision_FO.js]:
```suggestion
  4. Finish the order
```
For the sake of my OCD :grin:

File: tests/UI/campaigns/regression/currencies/computingPrecision_FO.js [hunk 4]
```diff
@@ -0,0 +1,535 @@
...
+  },
+);
+
+// Create sql query data to get last order discount and total price
+const sqlQueryData = {
+  name: 'Discount and TTC from last order',
+  sqlQuery: 'SELECT total_discounts, total_paid_tax_incl '
+    + 'from  ps_orders '
+    + 'ORDER BY id_order '
+    + 'DESC LIMIT 1;',
```
Comment (SimonGrn) [inline tests/UI/campaigns/regression/currencies/computingPrecision_FO.js]: Is there a way to make this query more resilient ? I'm thinking about launching differents campaigns in parallel and maybe having an order inserted at the wrong moment... Maybe adding the order reference that you get after creating it ?

File: tests/UI/campaigns/regression/currencies/computingPrecision_FO.js [hunk 5]
```diff
@@ -0,0 +1,535 @@
...
+      // Go to product page after searching its name
+      await homePage.searchProduct(page, orderToMake.product.name);
+      await searchResultsPage.goToProductPage(page, 1);
+
+      // Add the created product to the cart
+      await productPage.addProductToTheCart(page, orderToMake.quantityToOrder);
+
+      // Check cart page
+      const isCartPage = await cartPage.isCartPage(page);
+      await expect(isCartPage, 'Fail to go to cart page').to.be.true;
```
Comment (SimonGrn) [inline tests/UI/campaigns/regression/currencies/computingPrecision_FO.js]: What is this method `isCartPage` ? Is this specific to the `cartPage` class ? Is there a reason you didn't use the title check like we do for all the other pages (genuine question) ?

File: tests/UI/pages/BO/orders/index.js [hunk 6]
```diff
@@ -277,6 +277,17 @@ class Order extends BOBasePage {
     );
   }
 
+  /**
+   * Get order total price
+   * @param page
+   * @param row
+   * @return {number}
+   */
+  async getOrderTTCPrice(page, row) {
```
Comment (SimonGrn) [inline tests/UI/pages/BO/orders/index.js]: I'm pretty sure "TTC" is a french acronym :(

**Review by SimonGrn — COMMENTED** [review]
A few comments

File: tests/UI/pages/BO/orders/index.js (hunk 6 above)
Comment (NeOMakinG) [reply tests/UI/pages/BO/orders/index.js]: I think it's ATI (All Taxes Included)

File: tests/UI/campaigns/data/faker/cartRule.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/data/faker/cartRule.js:31]: When `freeShipping = 'on'`, tests created all cart rules with freeShipping, but it did not affect the total price since we choose free carrier in FO anyway.

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule.js (hunk 2 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule.js:151]: I just updated the name of the function in the cart rule page, so I have to do it in tests too, to avoid breaking them.
Same thing for tests identifiers to make it more appropriate.

File: tests/UI/campaigns/regression/currencies/computingPrecision_FO.js (hunk 5 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/regression/currencies/computingPrecision_FO.js]: Should be enough to check page title (I guess I was overthinking)

File: tests/UI/pages/BO/orders/index.js (hunk 6 above)
Comment (boubkerbribri) [reply tests/UI/pages/BO/orders/index.js]: It should be done for all function in this tests (and not breaking others)

File: tests/UI/campaigns/regression/currencies/computingPrecision_FO.js (hunk 4 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/regression/currencies/computingPrecision_FO.js]: Now we get order reference from BO order page to use it in sql query

File: tests/UI/campaigns/regression/currencies/computingPrecision_FO.js (hunk 3 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/regression/currencies/computingPrecision_FO.js]: done thanks

File: tests/UI/pages/BO/orders/index.js (hunk 6 above)
Comment (SimonGrn) [reply tests/UI/pages/BO/orders/index.js]: That's a problem ?

**Review by SimonGrn — DISMISSED** [review]

**Review by SimonGrn — APPROVED** [review]

File: tests/UI/pages/BO/orders/index.js (hunk 6 above)
Comment (boubkerbribri) [reply tests/UI/pages/BO/orders/index.js]: no it all good now for this PR

**Review by atomiix — APPROVED** [review]
